### PR TITLE
Implement Conversion Between OC and OTel Summary Metric

### PR DESCRIPTION
**Pull request by JasonXZLiu**
_Note:_ This PR is based off the Summary datapoint proto (that has not been merged upstream) in this [PR](https://github.com/open-telemetry/opentelemetry-proto/pull/227/files).

**Description:** 

This PR has two main parts: 

* Implement the [Summary datapoint proto](https://github.com/open-telemetry/opentelemetry-proto/pull/227/files) in the OTel Collector. Changes were made to _metrics_struct.go_ to create the generated Summary structs (using the `make genproto` and `make genpdata` commands).
* Add in the conversion between OpenCensus summary metrics and OTel Summary datapoints. This conversion directly adds compatibility for the Prometheus Summary metric in the PrometheusReceiver. 

**Link to tracking Issue:** [opentelemetry-specification/issues/1146](https://github.com/open-telemetry/opentelemetry-specification/issues/1146) and [opentelemetry-collector/issues/1638](https://github.com/open-telemetry/opentelemetry-collector/issues/1638)

A lot of the code in this PR is auto-generated from creating the structs and protos. The following files are were auto-generated:
* generated_metrics.go
* generated_metrics_test.go
* trace_config.pb.go 
* metrics.pg.go

**Testing:**

* Updated unit tests in _oc_to_metrics.go_ for conversion between OC summary metrics to OTel summary metrics
* Created end-to-end tests for the Summary metric in PrometheusReceiver

cc - @alolita

### CODE REVIEW

File: internal/data/testdata/metric.go
Comment (rakyll) [inline internal/data/testdata/metric.go:683]: Should we also set aggregation temporality to cumulative here?

**Review by rakyll — COMMENTED** [review]
Leaving the data conversion related review to the approvers. I left some high level questions and suggestions.

File: internal/data/testdata/metric.go
Comment (JasonXZLiu) [reply internal/data/testdata/metric.go:683]: We actually removed the aggregation temporality for the Summary metric in the proto (in this [PR](https://github.com/open-telemetry/opentelemetry-proto/pull/227)). The idea was that temporality does not make sense in the context of Summary (as it is more of an interval, rather than a delta or cumulative measurement).

File: consumer/pdata/metric.go
Comment (JasonXZLiu) [reply consumer/pdata/metric.go:181]: I may be wrong, but I believe OTLP has a representation for IntHistograms as there is instrumentation that specifically produces int values. Here, we are specifically concerned with backwards compatibility with Prometheus, so we only implemented the DoubleSummary.

That being said, I don't believe I will be able to implement the IntSummary here if it doesn't exist in the proto.

File: exporter/loggingexporter/logging_exporter_test.go
Comment (JasonXZLiu) [reply exporter/loggingexporter/logging_exporter_test.go:52]: Originally, `GenerateMetricsWithCountersHistograms` was used to generate a test for every metric type. I will change the name to be `GenerateMetricsAllTypesWithSampleDatapoints`.

File: internal/data/opentelemetry-proto-gen/metrics/v1/metrics.pb.go
Comment (tigrannajaryan) [inline internal/data/opentelemetry-proto-gen/metrics/v1/metrics.pb.go:65]: How did this file end up being modified? The content of this file is generated from internal/data/opentelemetry-proto and should not change unless the submodule in internal/data/opentelemetry-proto is updated.

I am updating opentelemetry-proto to 0.6.0 right now in https://github.com/open-telemetry/opentelemetry-collector/pull/2031.

To use the new summary data type you will need to first merge the PR that adds the summary data type, then make a new release of opentelemetry-proto and then use the newly released version here.

**Comment by bogdandrutu** [comment]
This needs to wait for the proto change to be merged.

**Review by tigrannajaryan — CHANGES_REQUESTED** [review]

File: internal/data/opentelemetry-proto-gen/metrics/v1/metrics.pb.go
Comment (JasonXZLiu) [reply internal/data/opentelemetry-proto-gen/metrics/v1/metrics.pb.go:65]: I originally copied the proto changes into the submodule and manually generated the proto to work on the rest of the implementation (while waiting on the proto changes to be merged). 

I will make a new release of the opentelemetry-proto and update this PR when the proto changes merged.

Comment (JasonXZLiu) [reply internal/data/opentelemetry-proto-gen/metrics/v1/metrics.pb.go:65]: Bogdan mentioned in the [proto PR](https://github.com/open-telemetry/opentelemetry-proto/commit/8ab21e9da6246e465cd9d50d405561aedef31a1e) that it isn't necessary to make a new release, but I could just point the submodule to the last commit. I believe this PR is updated with the proto changes now.

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
@tigrannajaryan do you suggest to not update now? This blocks some important feature for AWS.

**Review by tigrannajaryan — APPROVED** [review]

**Pull request merged**
